File: frontend/src/lib/api/dto/post-series.ts
export interface PostSeries {
  id: string;
  title: string;
  description?: string | null;
  cover?: string | null; // UUID
  created_at: string; // ISO 8601 timestamp
  last_updated: string; // ISO 8601 timestamp
}

export interface PostSeriesCreationDTO {
  title: string;
  description?: string | null;
  cover?: string | null; // UUID
}

export interface PostSeriesUpdateDTO {
  id: string;
  title: string;
  description?: string | null;
  cover?: string | null; // UUID
}

File: backend/app/api/routes/v1/dto/post_series.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class PostSeriesDTO(BaseModel):
    id: UUID
    title: str
    description: str | None = None
    cover: UUID | None = None
    created_at: datetime
    last_updated: datetime


class PostSeriesCreationDTO(BaseModel):
    title: str
    description: str | None = None
    cover: UUID | None = None


class PostSeriesMutationDTO(BaseModel):
    id: UUID
    title: str
    description: str | None = None
    cover: UUID | None = None
